src/sql/mir.rs

```diff
@@ -925,33 +1084,32 @@ impl SqlToMirConverter {
             }
 
 
-            // 2. Get columns used by each filter node.
-            let mut filter_columns = HashMap::new();
-            let mut filter_nodes = Vec::new();
-            let mut moved_filters = Vec::new();
+            // 2. Get columns used by each predicate. This will be used to check
+            // if we need to reorder predicates before group_by nodes.
+            let mut pred_cols = HashMap::new();
```

Comment:
Does this really contain columns? I read the code below as putting `ConditionExpression` objects from `qgn.predicates` into it.

Edit: Ah, it's a `HashMap<Column, Vec<ConditionExpression>>`, so it stores all predicates with each column contained in the predicates (an inverse mapping). Is it easy to avoid cloning all the predicates every time and to store a `Vec<&ConditionExpression>` instead?